src/sgl/core/bc_dds.cpp

```
@@ -0,0 +1,139 @@
+    SGL_CHECK(
+        bc_format.has_value(),
+        "bc_compressed_image_from_dds: DDS format is not a BC format (DXGI {})",
+        dds.dxgi_format()
+    );
+
+    BCCompressedImage result;
+    result.format = *bc_format;
+    result.mip_levels.resize(dds.mip_count());
+
+    for (uint32_t mip = 0; mip < dds.mip_count(); ++mip) {
+        uint32_t mip_width = mip >= 32 ? 1u : std::max(1u, dds.width() >> mip);
```

Cppcheck (2.21.0)

[style] The function 'resource_data' is never used.

(unusedFunction)